packages/nocodb/src/plugins/GenericS3/GenericS3.ts

```diff
@@ -14,21 +14,21 @@ import type { PutObjectRequest, S3 as S3Client } from '@aws-sdk/client-s3';
 import type { IStorageAdapterV2, XcFile } from '~/types/nc-plugin';
 import { generateTempFilePath, waitForStreamClose } from '~/utils/pluginUtils';
 
-interface GenerocObjectStorageInput {
+interface GenericObjectStorageInput {
   bucket: string;
   region?: string;
-  access_key: string;
-  access_secret: string;
+  access_key?: string;
+  access_secret?: string;
 }
 
 export default class GenericS3 implements IStorageAdapterV2 {
   public name;
 
   protected s3Client: S3Client;
-  protected input: GenerocObjectStorageInput;
+  protected input: GenericObjectStorageInput;
 
   constructor(input: unknown) {
-    this.input = input as GenerocObjectStorageInput;
+    this.input = input as GenericObjectStorageInput;
```

Comment:
it would be nice if you could use `constructor(protected input: GenericObjectStorageInput) {}`